std-bits/table/src/main/java/org/enso/table/data/column/builder/object/NumericBuilder.java

```diff
@@ -46,16 +49,16 @@ public void writeTo(Object[] items) {
     }
 
     @Override
-    public boolean canRetypeTo(long type) {
-        return !this.isDouble && type == Storage.Type.DOUBLE;
+    public boolean canRetypeTo(StorageType type) {
+        return !this.isDouble && Objects.equals(type, Constants.FLOAT_64);
```

Comment:
Why not?
```suggestion
        return !this.isDouble && Constants.FLOAT_64.equals(type);
```

Reply:
I would like to keep it as-is.

If I change it to the latter:
![image](https://user-images.githubusercontent.com/1436948/229093698-d6ba0bf0-c8dd-4a93-baff-260fa5fe646b.png)
I get a suggestion from IntelliJ that I could replace it with `==`.

This would be wrong as we do not guarantee that only a single instance of each storage type will exist - they are records that should be compared by-value and not by-reference.

I'd be worried that with this new form, someone may accidentally take the IDE's suggestion and replace with `==`. Worst part is - most of the functionality will still work as we _are_ mostly using the singletons. But some subtle places may break, so we would risk introducing a silent bug waiting to strike at an unexpected moment.

With current approach:
![image](https://user-images.githubusercontent.com/1436948/229094136-7d22df0c-4d2f-461e-8ca8-03ab60dc1946.png)
no wrongful suggestions are provided by the IDE. So it seems safer and more explicit that we need to use `equals`.

I don't know why IntelliJ provides that wrong suggestion in the first place. I'd guess maybe it sees an `UPPERCASE_IDENTIFIER` and thinks that it is a constant which can be compared on identity not equality. But that is not the case here.